<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>PreferencesDialog</class>
 <widget class="QDialog" name="PreferencesDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>550</width>
    <height>342</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Program preferences</string>
  </property>
  <property name="windowIcon">
   <iconset resource="icons.qrc">
    <normaloff>:/icons/process.png</normaloff>:/icons/process.png</iconset>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="0" column="0">
    <widget class="QListWidget" name="listPages">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Fixed" vsizetype="Expanding">
       <horstretch>1</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="maximumSize">
      <size>
       <width>130</width>
       <height>16777215</height>
      </size>
     </property>
     <property name="horizontalScrollBarPolicy">
      <enum>Qt::ScrollBarAlwaysOff</enum>
     </property>
     <property name="editTriggers">
      <set>QAbstractItemView::NoEditTriggers</set>
     </property>
     <property name="showDropIndicator" stdset="0">
      <bool>false</bool>
     </property>
     <property name="dragDropMode">
      <enum>QAbstractItemView::NoDragDrop</enum>
     </property>
     <property name="alternatingRowColors">
      <bool>false</bool>
     </property>
     <property name="selectionBehavior">
      <enum>QAbstractItemView::SelectRows</enum>
     </property>
     <property name="iconSize">
      <size>
       <width>32</width>
       <height>32</height>
      </size>
     </property>
     <property name="movement">
      <enum>QListView::Static</enum>
     </property>
     <property name="flow">
      <enum>QListView::TopToBottom</enum>
     </property>
     <property name="isWrapping" stdset="0">
      <bool>false</bool>
     </property>
     <property name="resizeMode">
      <enum>QListView::Adjust</enum>
     </property>
     <property name="spacing">
      <number>5</number>
     </property>
     <property name="viewMode">
      <enum>QListView::ListMode</enum>
     </property>
     <property name="uniformItemSizes">
      <bool>true</bool>
     </property>
     <property name="wordWrap">
      <bool>true</bool>
     </property>
     <property name="selectionRectVisible">
      <bool>false</bool>
     </property>
     <property name="currentRow">
      <number>-1</number>
     </property>
     <property name="sortingEnabled">
      <bool>false</bool>
     </property>
     <item>
      <property name="text">
       <string>General</string>
      </property>
      <property name="icon">
       <iconset resource="icons.qrc">
        <normaloff>:/icons/computer.png</normaloff>:/icons/computer.png</iconset>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Sound</string>
      </property>
      <property name="icon">
       <iconset resource="icons.qrc">
        <normaloff>:/icons/sound.png</normaloff>:/icons/sound.png</iconset>
      </property>
     </item>
     <item>
      <property name="text">
       <string>DTW</string>
      </property>
      <property name="icon">
       <iconset resource="icons.qrc">
        <normaloff>:/icons/dtw_chart.png</normaloff>:/icons/dtw_chart.png</iconset>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Recognizing</string>
      </property>
      <property name="icon">
       <iconset resource="icons.qrc">
        <normaloff>:/icons/search.png</normaloff>:/icons/search.png</iconset>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Charts</string>
      </property>
      <property name="icon">
       <iconset resource="icons.qrc">
        <normaloff>:/icons/chart.png</normaloff>:/icons/chart.png</iconset>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Appearance</string>
      </property>
      <property name="icon">
       <iconset resource="icons.qrc">
        <normaloff>:/icons/application_edit.png</normaloff>:/icons/application_edit.png</iconset>
      </property>
     </item>
    </widget>
   </item>
   <item row="0" column="1" colspan="2">
    <widget class="QStackedWidget" name="stackedWidget">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
       <horstretch>5</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="pageGeneral">
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QGroupBox" name="gbxGeneral">
         <property name="title">
          <string>General configuration</string>
         </property>
         <layout class="QFormLayout" name="formLayout_6">
          <item row="0" column="0">
           <widget class="QLabel" name="lblVerbosity">
            <property name="text">
             <string>Message verbosity:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QComboBox" name="cbxVerbosity">
            <property name="currentIndex">
             <number>1</number>
            </property>
            <item>
             <property name="text">
              <string>Low</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Medium</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>High</string>
             </property>
            </item>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="pageSound">
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <widget class="QGroupBox" name="gbxSound">
         <property name="title">
          <string>Sound settings</string>
         </property>
         <layout class="QFormLayout" name="formLayout">
          <item row="0" column="0">
           <widget class="QLabel" name="lblFrameLength">
            <property name="text">
             <string>Frame length:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QSpinBox" name="editFrameLength">
            <property name="suffix">
             <string> ms</string>
            </property>
            <property name="minimum">
             <number>5</number>
            </property>
            <property name="maximum">
             <number>600</number>
            </property>
            <property name="value">
             <number>20</number>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="lblWindowType">
            <property name="text">
             <string>Window function type:</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QComboBox" name="cbxWindowType">
            <property name="currentIndex">
             <number>0</number>
            </property>
            <item>
             <property name="text">
              <string>Rectangular</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Hamming</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Hann</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Blackman</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Barlett</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Flat-top</string>
             </property>
            </item>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="lblPreemphasis">
            <property name="text">
             <string>Preemphasis factor:</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QDoubleSpinBox" name="editPreemphasis">
            <property name="decimals">
             <number>4</number>
            </property>
            <property name="maximum">
             <double>1.000000000000000</double>
            </property>
            <property name="singleStep">
             <double>0.001000000000000</double>
            </property>
            <property name="value">
             <double>0.937500000000000</double>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="lblFrameOverlap">
            <property name="text">
             <string>Frame overlap:</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QDoubleSpinBox" name="editFrameOverlap">
            <property name="maximum">
             <double>1.000000000000000</double>
            </property>
            <property name="singleStep">
             <double>0.010000000000000</double>
            </property>
            <property name="value">
             <double>0.660000000000000</double>
            </property>
           </widget>
          </item>
          <item row="4" column="1">
           <widget class="QPushButton" name="btnMelFilters">
            <property name="text">
             <string>Choose enabled Mel filters...</string>
            </property>
           </widget>
          </item>
          <item row="4" column="0">
           <widget class="QLabel" name="lblMelFilters">
            <property name="text">
             <string>Mel filters:</string>
            </property>
           </widget>
          </item>
          <item row="5" column="0">
           <widget class="QLabel" name="lblParamsPerFrame">
            <property name="text">
             <string>Parameters per frame:</string>
            </property>
           </widget>
          </item>
          <item row="5" column="1">
           <widget class="QSpinBox" name="editParamsPerFrame">
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="maximum">
             <number>25</number>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="pageDtw">
      <layout class="QGridLayout" name="gridLayout_4">
       <item row="0" column="0">
        <widget class="QGroupBox" name="gbxDtw">
         <property name="title">
          <string>DTW settings</string>
         </property>
         <layout class="QFormLayout" name="formLayout_4">
          <item row="0" column="0">
           <widget class="QLabel" name="lblDistanceType">
            <property name="text">
             <string>Distance type:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QComboBox" name="cbxDistanceType">
            <item>
             <property name="text">
              <string>Euclidean</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Manhattan</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Chebyshev</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Minkowski</string>
             </property>
            </item>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="lblNormalizationType">
            <property name="text">
             <string>Normalization type:</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QComboBox" name="cbxNormalizationType">
            <item>
             <property name="text">
              <string>No normalization</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Diagonal</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Sum of sides</string>
             </property>
            </item>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="lblPassType">
            <property name="text">
             <string>Pass type:</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QComboBox" name="cbxPassType">
            <item>
             <property name="text">
              <string>Neighbors</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Diagonals</string>
             </property>
            </item>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="pageRecognizing">
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <item>
        <widget class="QGroupBox" name="gbxRecognizing">
         <property name="title">
          <string>Recognizing settings</string>
         </property>
         <layout class="QFormLayout" name="formLayout_5">
          <property name="fieldGrowthPolicy">
           <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
          </property>
          <item row="0" column="0">
           <widget class="QLabel" name="lblMinPatternSize">
            <property name="text">
             <string>Minimal pattern size:</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="lblMaxPatternSize">
            <property name="text">
             <string>Maximal pattern size:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QDoubleSpinBox" name="editMinPatternSize">
            <property name="suffix">
             <string>%</string>
            </property>
            <property name="decimals">
             <number>0</number>
            </property>
            <property name="value">
             <double>60.000000000000000</double>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QDoubleSpinBox" name="editMaxPatternSize">
            <property name="suffix">
             <string>%</string>
            </property>
            <property name="decimals">
             <number>0</number>
            </property>
            <property name="minimum">
             <double>100.000000000000000</double>
            </property>
            <property name="maximum">
             <double>666.000000000000000</double>
            </property>
            <property name="value">
             <double>140.000000000000000</double>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="pageCharts">
      <layout class="QGridLayout" name="gridLayout_2">
       <item row="0" column="0">
        <widget class="QGroupBox" name="gbxCharts">
         <property name="title">
          <string>Charts settings</string>
         </property>
         <layout class="QFormLayout" name="formLayout_2">
          <item row="0" column="0" colspan="2">
           <widget class="QCheckBox" name="cbxChartsOverlay">
            <property name="text">
             <string>Add text overlay with detailed information?</string>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="lblDefaultImageType">
            <property name="text">
             <string>Default image format for saving:</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QComboBox" name="cbxDefaultImageType">
            <item>
             <property name="text">
              <string>.png</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>.jpg</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>.bmp</string>
             </property>
            </item>
           </widget>
          </item>
          <item row="2" column="0" colspan="2">
           <widget class="QCheckBox" name="cbxDrawDtwAccumulated">
            <property name="text">
             <string>Draw DTW accumulated distances?</string>
            </property>
            <property name="checked">
             <bool>false</bool>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="pageAppearance">
      <layout class="QGridLayout" name="gridLayout_3">
       <item row="0" column="0">
        <widget class="QGroupBox" name="gbxAppearance">
         <property name="title">
          <string>Appearance</string>
         </property>
         <layout class="QFormLayout" name="formLayout_3">
          <item row="0" column="0" colspan="2">
           <widget class="QCheckBox" name="cbxUseStylesheet">
            <property name="text">
             <string>Use stylesheets?</string>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="lblStylesheet">
            <property name="text">
             <string>Stylesheet name:</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QComboBox" name="cbxStylesheet">
            <item>
             <property name="text">
              <string>default</string>
             </property>
            </item>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="lblWaveFormColor">
            <property name="text">
             <string>Waveform chart color:</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="ColorWidget" name="colorWaveform" native="true"/>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item row="1" column="0" colspan="2">
    <spacer name="horizontalSpacer">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>328</width>
       <height>20</height>
      </size>
     </property>
    </spacer>
   </item>
   <item row="1" column="2">
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok|QDialogButtonBox::RestoreDefaults</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>ColorWidget</class>
   <extends>QWidget</extends>
   <header>widgets/color/ColorWidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="icons.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>PreferencesDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>540</x>
     <y>332</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>PreferencesDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>540</x>
     <y>332</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>listPages</sender>
   <signal>currentRowChanged(int)</signal>
   <receiver>stackedWidget</receiver>
   <slot>setCurrentIndex(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>77</x>
     <y>238</y>
    </hint>
    <hint type="destinationlabel">
     <x>244</x>
     <y>248</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cbxUseStylesheet</sender>
   <signal>toggled(bool)</signal>
   <receiver>cbxStylesheet</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>309</x>
     <y>59</y>
    </hint>
    <hint type="destinationlabel">
     <x>494</x>
     <y>85</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
